src/lsm/table_iterator.zig

```diff
@@ -185,15 +182,15 @@ pub fn TableIteratorType(comptime Table: type, comptime Storage: type) type {
             assert(it.read_table_index);
             it.read_table_index = false;
 
+            // Copy the bytes read into a buffer owned by the iterator since the Grid
+            // only guarantees the provided pointer to be valid in this callback.
+            mem.copy(u8, it.index_block, block);
+
```

Comment:
Just out of curiosity, why did this move?

Reply:
I moved it above the `index_block_callback` just to keep with the invariant that the caller is done with the index block before the callback.